dev/fattest.simplicity/src/componenttest/custom/junit/runner/RepeatTestFilter.java

```diff
@@ -58,7 +59,8 @@ public static boolean shouldRun(FrameworkMethod method) {
 
         String[] skipValues = MultivalueSkips.getSkipForRepeatValues(anno.value());
         for (String action : skipValues) {
-            if (repeatStackContainsActionByID(action)) {
+            //Skipping for CHECKPOINT_RULE is currently possible only for methods since the isActive() never returns true on class level. It will skip running the checkpoint repeat for any method when the checkpoint rule is added as a ClassRule.
+            if (repeatStackContainsActionByID(action) || (action.equals(CheckpointRule.ID) && CheckpointRule.isActive())) {
```

Comment:
Consider extracting the skip for checkpoint logic to another method so this can be something like:
```java
        if (repeatStackContainsActionByID(action) || checkpointActionActive(action)) {
```

Reply:
Done!